pulsar-client-cpp/lib/MessageImpl.h

```diff
@@ -43,13 +43,24 @@ class MessageImpl {
     SharedBuffer payload;
     MessageId messageId;
     ClientConnection* cnx_;
+    std::string topicName_;
```

Comment:
As an optimization, to avoid to create (and copy) the topic string object for each message id, we could use a `std::string*` to store the name. This would work if we pass a reference of the topic name held by the consumer instance, which will be valid. 

The `getTopicName()` would still have the same signature.